opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/aggregate.py

```diff
@@ -148,12 +154,18 @@ def take_checkpoint(self):
 
     def merge(self, other):
         self.mmsc.merge(other.mmsc)
+        last = self.checkpoint.last
+        if (other.last_update_timestamp is not None and
+            self.last_update_timestamp is not None):
+            if self.last_update_timestamp < other.last_update_timestamp:
+                self.last_update_timestamp = other.last_update_timestamp
+                last = other.checkpoint.last
+        elif self.last_update_timestamp is None:
+            self.last_update_timestamp = other.last_update_timestamp
+            last = other.checkpoint.last
         self.checkpoint = self._TYPE(
             *(
                 self.mmsc.checkpoint
-                + (other.checkpoint.last or self.checkpoint.last or 0,)
+                + (last,)
```

Comment:
It'd be really nice to have a test checking that `last` is chosen according to the timestamp.